Shuttle-back/src/main/java/com/shuttle/driver/DriverController.java

```diff
@@ -124,17 +143,23 @@ public ResponseEntity<WorkHours> changeWorkHours(@PathVariable(value = "working-
 
     }
 
+    @PermitAll
     @GetMapping("/api/driver/{id}/ride")
-    public ResponseEntity<ListDTO<RideDTO>> getRideHistory(@PathVariable(value = "id") Long id,
-                                                           @PathParam("page") int page, @PathParam("size") int size,
-                                                           @PathParam("from") String from, @PathParam("to") String to,
-                                                           @PathParam("to") String sort) {
-        ListDTO<RideDTO> rideDTOListDTO = new ListDTO<>();
-        rideDTOListDTO.setTotalCount(page);
-        List<RideDTO> rideDTOList = new ArrayList<>();
-        rideDTOList.add(new RideDTO());
-        rideDTOListDTO.setResults(rideDTOList);
-        return new ResponseEntity<>(rideDTOListDTO, HttpStatus.OK);
+    public ResponseEntity<ListDTO<Ride>> getRideHistory(@PathVariable(value = "id") Long id,
+                                                     @PathParam("page") int page, @PathParam("size") int size,
+                                                     @PathParam("from") String from, @PathParam("to") String to,
+                                                     @PathParam("to") String sort) {
+        Pageable pageParams =
+                PageRequest.of(page, size, Sort.by(sort));
+
+        DateTimeFormatter formatter = DateTimeFormatter.ISO_INSTANT;
+        LocalDateTime startTime = LocalDateTime.ofInstant(Instant.from(formatter.parse(from)), ZoneId.of(ZoneOffset.UTC.getId()));
+        LocalDateTime endTime = LocalDateTime.ofInstant(Instant.from(formatter.parse(to)), ZoneId.of(ZoneOffset.UTC.getId()));
+        Page<Ride> rides = rideRepository.getAllBetweenDates(startTime, endTime, id, pageParams);
```

Comment:
In the future, we'll need to check whether `id` is valid (otherwise throw 400 bad request) and whether a driver with that ID exists (otherwise throw 404 not found). Those should be done here, and for consistency sake you _could_ send the entire `Driver` object to this function instead of `id`.

Reply:
Also, this call should be sent to `RideService` which would delegate it to `RideRepository`.